blocks/api/factory.js

```diff
@@ -21,15 +21,17 @@ import { __ } from '@wordpress/i18n';
  * Internal dependencies
  */
 import { getBlockType } from './registration';
+import { getBlockAttributes } from './parser';
 
 /**
  * Returns a block object given its type and attributes.
  *
- * @param  {String} name             Block name
- * @param  {Object} blockAttributes  Block attributes
- * @return {Object}                  Block object
+ * @param  {String}  name             Block name
+ * @param  {Object}  blockAttributes  Block attributes
+ * @param  {?String} innerHTML        Block HTML
+ * @return {Object}                   Block object
  */
-export function createBlock( name, blockAttributes = {} ) {
+export function createBlock( name, blockAttributes = {}, innerHTML ) {
```

Comment:
Before this change, `createBlock` meant something very basic: run a simple check against `blockType.attributes` to grab the default values when values are missing, and return a plain object representing a block. No parsing is involved, etc.

1. Are we OK with changing that?
2. Do we prefer adding a dedicated function, e.g. `createBlockFromHTML( name, innerHTML, extraAttributes )`?
3. Do we prefer not adding anything and requiring authors to combine `createBlock` and `getBlockAttributes` on their own?

Reply:
I feel bad not mentioning this in the PR description. Yes, there are some options here, and code wise all of them don't require much extra. I first used `getBlockAttributes`, but then thought it made sense as part of createBlock. A block usually consists of a name, attributes and HTML, so it makes sense to be able to create a block from all that information?

Reply:
Could simplify things internally as well. In parser.js we do:

```js
const block = createBlock(
	 name,
	getBlockAttributes( blockType, innerHTML, attributes )
);
```

So why not:

```js
const block = createBlock( name, attributes, innerHTML );
```

Reply:
I can understand where this idea comes from. We never used `getBlockAttributes` outside the `api` yet I think.

But I think I'll be in favor of keeping the `createBlock` simple. And compose the two functions when needed because there's a major difference in the expected "attributes" argument here.

the attributes parameter of `createBlock` can be any attribute allowed by the block, it behaves like "defaultAttributes" somehow. While the "attributes" parameter of `getBlockAttributes` represent only the "comment" attributes (the parsed JSON).

Reply:
See also #3745, conflicting changes with API and intending to pass inner _children_ nodes.